hw/ip_templates/pinmux/data/pinmux.hjson.tpl

```diff
@@ -572,19 +572,27 @@
                   regwen:       "MIO_PAD_ATTR_REGWEN",
                   regwen_multi: "true",
                   cname:        "MIO_PAD",
-                  resval:       0
                   fields: [
                     { bits: "0",
                       name: "invert",
                       desc: "Invert input and output levels."
+                      resval: 0
                     },
                     { bits: "1",
                       name: "virtual_od_en",
                       desc: "Enable virtual open drain."
+                      resval: 0
                     },
                     { bits: "2",
                       name: "pull_en",
                       desc: "Enable pull-up or pull-down resistor."
+                      // Note: The special index where this has a different resval depends on the
+                      // TargetCfg parameter that gets passed in. That, in turn, is configured at
+                      // chip level and it would be nicer if we could resolve it as we expanded this
+                      // template. Instead, we expect the top that renders the template to give
+                      // matching parameters and we include a basic check in pinmux.sv.tpl to make
+                      // sure everything is in sync.
+                      resval: (multireg_idx == ${tap_strap0_idx})
```

Comment:
I think this code should only be rendered if `tap_strap0_idx ` > 0

Reply:
Honestly, I'm really unconvinced that this PR has the "right way" to solve the problem! I think the better approach is probably a bodge on the UVM side (where we squirt in some information to tell the register model the correct resval to expect)

I'm going on holiday next week, but intend to come back to this and make the change more carefully. That's part of the reason that I factored out the "obviously right" bit into #27289.